Library/brew.rb

```diff
@@ -58,6 +58,10 @@ def require?(path)
   ENV["PATH"] += "#{File::PATH_SEPARATOR}#{HOMEBREW_ENV_PATH}/scm"
 
   if cmd
+    if cmd == "cask" && (HOMEBREW_CELLAR/"brew-cask").exist?
+      system(HOMEBREW_BREW_FILE, "uninstall", "--force", "brew-cask")
+    end
```

Comment:
I think this should be moved further down (below the help handling) to bring it closer to where we actually dispatch to an external command.